vulture_os/toolkit/api_parser/waf_barracuda/waf_barracuda.py

```diff
@@ -0,0 +1,129 @@
+#!/home/vlt-os/env/bin/python
+"""This file is part of Vulture OS.
+
+Vulture OS is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "Gaultier Parain"
+__credits__ = ["Kevin Guillemot"]
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'WAF BARRACUDA API'
+__parser__ = 'WAF BARRACUDA'
+
+import json
+import logging
+import requests
+
+from datetime import timedelta
+from django.utils import timezone
+from django.conf import settings
+from toolkit.api_parser.api_parser import ApiParser
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+class WAFBarracudaAPIError(Exception):
+    pass
+
+class WAFBarracudaParser(ApiParser):
+
+    HEADERS = {
+        "Content-Type": "application/json",
+        'Accept': 'application/json'
+    }
+    WAF_BARRACUDA_HOST = "api.waas.barracudanetworks.com"
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.waf_barracuda_token = data["waf_barracuda_token"]
+
+        self.session = None
+
+    def __connect(self):
+
+        if not self.session:
+            self.session = requests.Session()
+            self.session.proxies = self.proxies
+            self.session.headers.update(self.HEADERS)
+            self.session.headers.update({"Authorization": f"Bearer {self.waf_barracuda_token}"})
+
+    def __execute_query(self, url, query={}, timeout=15):
+        response = self.session.get(url,
+            params=query,
+            headers=self.HEADERS,
+            timeout=timeout,
+            proxies=self.proxies,
+            verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl
+        )
+        if response.status_code != 200:
+            raise WAFBarracudaAPIError(f"{[__parser__]}:get_logs: Status code = {response.status_code}, Error = {response.text}")
+        else:
+            return response.json()
+
+    def get_logs(self, since, to):
+        self.__connect()
+        result = []
+        #itemsPerPage max is 1000
+        query = {"from": since.timestamp(), "to": to.timestamp(), "itemsPerPage": "1000"}
```

Comment:
I see a `itemsPerPage` argument, but no page management in the script!
what if we get more than 1k logs in the queried interval ?!